src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/modifier/modifierorder/InputModifierOrderSealedAndNonSealedNoViolation.java

```diff
@@ -0,0 +1,35 @@
+//non-compiled with javac: Compilable with Java15
+package com.puppycrawl.tools.checkstyle.checks.modifier.modifierorder;
+
+/* Config:
+ * default
+ */
+
+public sealed class InputModifierOrderSealedAndNonSealedNoViolation // ok
+    permits Circle, Square, Rectangle {
+}
+
+final class Circle extends InputModifierOrderSealedAndNonSealedNoViolation implements Squircle {
+}
+
+sealed class Rectangle extends InputModifierOrderSealedAndNonSealedNoViolation // ok
+    implements Cloneable
+    permits TransparentRectangle, FilledRectangle {
+}
+
+final class TransparentRectangle extends Rectangle {
+}
+
+sealed class Square extends InputModifierOrderSealedAndNonSealedNoViolation implements Squircle {
+    private sealed class OtherSquare extends Square permits OtherSquare2 { // ok
+    }
+
+    private final class OtherSquare2 extends OtherSquare {
+    }
+}
+
+final strictfp class FilledRectangle extends Rectangle {
+}
+
+sealed interface Squircle permits Circle, Square { // ok
```

Comment:
can these classes be static? if yes, please add it to test cases

Reply:
Done.